node_modules/@stdlib/iter/cartesian-power/lib/main.js

```diff
@@ -0,0 +1,147 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+var setReadOnly = require('@stdlib/utils/define-nonenumerable-read-only-property');
+var isNumber = require('@stdlib/assert/is-number').isPrimitive;
+var isnan = require('@stdlib/math/base/assert/is-nan');
+var isNonNegativeInteger = require('@stdlib/assert/is-nonnegative-integer').isPrimitive;
+var iteratorSymbol = require('@stdlib/symbol/iterator');
+var format = require('@stdlib/string/format');
+
+// MAIN //
+
+/**
+* Returns an iterator which generates the Cartesian power of an input array-like object.
+*
+* @param {Array|TypedArray|Iterable} x - Input array-like object.
+* @param {number} n - Integer power.
+* @throws {TypeError} first argument must be an array-like object
+* @throws {TypeError} second argument must be a nonnegative integer
+* @returns {Iterator} iterator
+*
+* @example
+* var iterCartesianPower = require( '@stdlib/iter/cartesian-power' );
+*
+* var arr = ['a', 'b', 'c'];
+* var n = 2;
+* var iterator = iterCartesianPower(arr, n);
+*
+* var result;
+* while (!(result = iterator.next()).done) {
+*     console.log(result.value);
+* }
+*/
+
+function iterCartesianPower(x, n) {
+    var iter;
+    var FLG = false;
+    var i = 0;
+    var j = [];
+    var idx = [];
+
+    // Validate input arguments
+    if (!Array.isArray(x) && !isTypedArray(x) && !isIterable(x)) {
```

Comment:
Please have a look through the codebase and match our spacing conventions. Additionally, you should not be checking for iterables, as you are not handling iterables in the implementation. Lastly, check for a collection, rather than using `isArray` and `isTypedArray`.